File: py/dataquery_client.py
import grpc
import dataquery_pb2
import dataquery_pb2_grpc


###########################################################
# Entry Point Run
def run_client():
    grpc_server_str = "localhost:9999"

    with grpc.insecure_channel(grpc_server_str) as channel:
        stub = dataquery_pb2_grpc.DataQueryStub(channel)

        user_id = "victor.ty.ho"
        user_token = "wqeadsAATYGDGDqwwqeadsAATYGDGDqw"
        data_id = "ZN"
        data_type = "snap.NKY"
        return_type = 0

        # 1. Client calls GetData()
        get_data_resp = stub.GetData(dataquery_pb2.GetDataRequest(user_id=user_id,
                                                                  user_token=user_token,
                                                                  data_id=data_id,
                                                                  data_type=data_type,
                                                                  return_type=return_type))
        print("Python Client, GetData response. Code= %d, Payload= %s" %
              (get_data_resp.response_code, get_data_resp.data_payload))

        # 2. Client calls InspectMetaData()
        meta_data_resp = stub.InspectMetadata(dataquery_pb2.MetadataRequest(user_id=user_id,
                                                                            user_token=user_token,
                                                                            data_id=data_id,
                                                                            data_type=data_type))
        print("Python Client, InspectMetadata response. Code= %d, MetaData= %s" %
              (meta_data_resp.response_code, str(meta_data_resp.meta_data)))


if __name__ == "__main__":
    run_client()

File: rs/src/dataquery-client.rs
use dataquery::data_query_client::DataQueryClient;
use dataquery::GetDataRequest;
use dataquery::MetadataRequest;
//#[cfg(test)]
//use mockall::{automock, predicate::*};

// new module to include types based on proto file
pub mod dataquery {
    tonic::include_proto!("dataquery");
}


fn create_get_data_request() -> GetDataRequest {
    let req = GetDataRequest {
        user_id: "victor.ty.ho".to_owned(),
        user_token: "wqeadsAATYGDGDqwwqeadsAATYGDGDqw".to_owned(),
        data_id:  "ZN".to_owned(),
        data_type: "snap.NKY".to_owned(),
        return_type: 0,  
    };
    return req;
}

fn create_meta_data_request() -> MetadataRequest {
    let req = MetadataRequest {
        user_id: "victor.ty.ho".to_owned(),
        user_token: "wqeadsAATYGDGDqwwqeadsAATYGDGDqw".to_owned(),
        data_id:  "JGB10Y".to_owned(),
        data_type: "snap.TSE".to_owned(),
    };
    return req;
}

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut client = DataQueryClient::connect("http://[::1]:9999").await?;

    // #1 RPC call to get_data()  
    let data_req = tonic::Request::new(create_get_data_request());
    let response = client.get_data(data_req).await?;
    println!("get_data RESPONSE={:?}", response);

    // #2 RPC call to inspect_metadata()
    let meta_req = tonic::Request::new(create_meta_data_request());
    let response2 = client.inspect_metadata(meta_req).await?;
    println!("inspect_metadata RESPONSE={:?}", response2);

    Ok(())
}


//////////////////////////////////////////////////////////////////////
// Unit Tests (live in same file as code)
#[cfg(test)]
mod client_tests {
    use super::*;

    #[test]
    fn get_data_req_test() {
        let req = create_get_data_request();
        assert!(!req.user_id.is_empty());
    }

    #[test]
    fn meta_data_req_test() {
        let req = create_meta_data_request();
        assert!(!req.user_id.is_empty());
    }
}